boostedblob/azure_auth.py

```diff
@@ -174,9 +166,31 @@ async def get_access_token(cache_key: Tuple[str, Optional[str]]) -> Tuple[Any, f
             return (auth, now + AZURE_SHARED_KEY_EXPIRATION_SECONDS)
         raise RuntimeError(
             f"Found storage account key, but it was unable to access storage account: '{account}'"
+        ) 
+    
+    from .globals import config
+
+    if creds["_azure_auth"] == "msi":
+        # Managed Service Identity
+        req = create_access_token_request(
+            creds=creds,
+            scope=f"https://{account}.blob.core.windows.net/.default",
+            success_codes=(200, 400),
         )
+        
+        async with req.execute() as resp:
```

Comment:
you don't handle the 400?

Reply:
yeah i'll make sure the server can handle this